### ISSUE TITLE
RichTextBox always sends its entire contents as an accessibility event on focus.

### ISSUE THREAD

**Issue by trypsynth** [dotnet/winforms#12772]
### .NET version

DotNet 9.0

### Did it work in .NET Framework?

Not tested/verified

### Issue description

If I create a RichTextBox, alt+tab away from the window, and alt+tab back, both JAWS and NVDA report the entire contents of the RichTextBox as its own accessibility event. This is not standard behavior.
Oddly enough, narrator is not effected, reading nothing at all when the control takes focus.
Form1.cs:
```
namespace RichEditTest;

public class Form1 : Form {
	private RichTextBox textBox;

	public Form1() {
		this.Text = "RichEdit Test";
		this.Width = 450;
		this.Height = 250;
		textBox = new RichTextBox();
		textBox.AccessibleName = "Content";
		textBox.Text = "This is a test\r\nthat spans multiple\r\nlines.";
		textBox.Size = new Size(400, 200);
		textBox.Location = new Point(10, 10);
		this.Controls.Add(textBox);
	}
}
```
Program.cs
```
namespace RichEditTest;

static class Program
{
    /// <summary>
    ///  The main entry point for the application.
    /// </summary>
    [STAThread]
    static void Main()
    {
        // To customize application configuration such as set high DPI settings or default font,
        // see https://aka.ms/applicationconfiguration.
        ApplicationConfiguration.Initialize();
        Application.Run(new Form1());
    }    
}
```

### Steps to reproduce

1. Run the provided code with NVDA or JAWS active.
2. Alt+tab away from the window and back.
3. Note that the entire content gets read, not just the currently focused line. Seemingly it's being sent as its own accessibility event.

**Comment by elachlan** [comment]
@Olina-Zhang  can your team please test?

**Comment by Olina-Zhang** [comment]
Tested it in .NET versions and .NET framework 4.8.1, they have same result:
For NVDA, when alt+tab away from the window and back, it will announce the first line content, but Narrator doesn't announce.

NVDA:

![Image](https://github.com/user-attachments/assets/794eba2f-3b3c-4174-913e-3bf47293b707)

Narrator:

![Image](https://github.com/user-attachments/assets/c891642d-35fc-48e1-8482-3b0eb2f27847)

**Comment by Zheng-Li01** [comment]
Below are the results for MFC project between Narrator and NVDA tools.

![Image](https://github.com/user-attachments/assets/ba58492a-b8ba-419a-91ae-39e9b858cf5f)

**Comment by merriemcgaw** [comment]
@TheQuinbox thank you so much for filing this issue! This is a great find. That said, it looks like this is an issue with the RichEdit control itself, rather than WinForms specifically. Unfortunately, I think one would need to file an issue with the Office team, but I don't recall if there is any active development of the control. 

@TheQuinbox - could you file a feedback ticket with Windows (because it is distributed with Windows) and if you can provide a link here, we can do our best to pull together a case for someone to take a look at the issue. I'm going to close this from our end, but we want to make sure the right owners are notified of the issue.

**Comment by TheQuinbox** [comment]
Hello,
Thanks for your reply! I don't think this is an issue with the rich text box control itself, because there are applications I've seen use it without this behavior whatsoever. For example, https://github.com/samtupy/Pipe2Textbox. That's a very small C program with no CRT that links against the Win32 API only, so it should show you pretty trivially.
Thanks again!

**Comment by TheQuinbox** [comment]
Hi @merriemcgaw,
Are there any updates on this issue? Plans to reopen it/etc, at the very least? Unless you can provide me with a test case to confirm this is indeed specific to the rich  edit control in native Win32. I don't find this likely though as rich edit controls in both wxWidgets and native Win32 C don't exhibit this behavior.
Thanks for your time,
Quin.

**Comment by merriemcgaw** [comment]
@TheQuinbox Thank you for pushing back, you made us look more closely. And, of course, you are correct. We've pinpointed where we are [sending the entire contents](https://github.com/dotnet/winforms/blob/f79026644d3f38a7ab9cfb734a011d5448346d66/src/System.Windows.Forms/src/System/Windows/Forms/Controls/RichTextBox/RichTextBox.cs#L2400-L2413). during the `OnGotFocus()` event. 

It looks like this code was added several years ago in response to nothing being sent to Narrator when getting focus, which is a clear violation of accessibility guidelines. Obviously sending then entire contents of the RichTextBox every time the control gets focus is not performant or reasonable when you've got a lot of text. We'll look on our side at what MFC exactly does, but I'm interested from the assistive technology user perspective, what would you find appropriate to be sending? If you have a large RichTextBox with lots of content, what is it that you would want announced by NVDA/Jaws/Narrator? I think we have the potential here to optimize for both performance and user experience. 

@LeafShi1 can your team take a look at the sources for the MFC control and see what it is sending when it gets focus? At the very least we can mimic that behavior. I'd like to see this in .NET 10 for sure.

**Comment by TheQuinbox** [comment]
Hi @merriemcgaw,
Thanks for your reply and your time looking into this!
For me, what I would like announced is the line with the current focus, as well as the standard properties of the control such as its read-only state. For example, if I open the  QRead application from https://getaccessibleapps.com with NVDA and focus a document, then tab away from the rich text field and back to it, this is what happens. Similarly, in the Win32 Pipe2Textbox program I linked above, this is what happens.
Interestingly, testing both of these applications with Narrator confirms that Narrator not reading any text when the control gets focus is not limited to Winforms at all, so I'd say that's the bug that has to be fixed not by Winforms, but by the Narrator/control developers.
Thanks again for your time and quick replies! :)

**Comment by JeremyKuhne** [comment]
> For me, what I would like announced is the line with the current focus, as well as the standard properties of the control such as its read-only state.

This seems reasonable to me. :)

**Comment by LeafShi1** [comment]
[like]  Leaf Shi (BEYONDSOFT CONSULTIN... reacted to your message:
________________________________

**Comment by merriemcgaw** [comment]
@LeafShi1 let's chat about what we want to send to Narrator and UIA when we meet next.

**Comment by LeafShi1** [comment]
RichTextBox always sends its entire content, due to changes in PR #7389

**Comment by merriemcgaw** [comment]
There is some risk in the implementation options available to us. After talking with our internal accessibility team, it seems that there is an easy way for the screen reader user to "hush" the reading once focus is received. In that case we don't have a compliance issue, and with the risk of making changes here it's better to let the additional content be sent to the AT.
